### PR TITLE
Fix Violation.to_github_annotation method

### PR DESCRIPTION
**Pull request by kamalq97**
Took inspiration from this discussion: [Annotations, how to create them? #26703](https://github.com/orgs/community/discussions/26703)

### CODE REVIEW

File: riff/violation.py [hunk 1]
```diff
@@ -32,13 +32,7 @@ class Violation(NamedTuple):
...
     def to_github_annotation(self: "Violation") -> str:
         relativized_path = self.path.relative_to(Path.cwd())
-        first_part = f"::error title=Ruff ({self.error_code}),file={self.path},line={self.line_start},"
-        "col={self.column_start},endLine={self.line_end},endColumn={self.column_end}::"
-
-        second_part = f"{relativized_path}:{self.line_start}:"
-        if self.column_start:
-            second_part += f"{self.column_start}:"
-        return f"{first_part}{second_part} {self.error_code} {self.message}"
+        return f'::error file={relativized_path},line={self.line_start},col={self.column_start},endLine={self.line_end},endColumn={self.column_end}::"Ruff ({self.error_code})"'
```
Comment (dorschw) [inline riff/violation.py:35]: what if there's no `self.column_start`?

File: riff/violation.py [hunk 2]
```diff
@@ -30,15 +30,21 @@ class Violation(NamedTuple):
...
+        
         if self.column_start:
-            second_part += f"{self.column_start}:"
-        return f"{first_part}{second_part} {self.error_code} {self.message}"
+            annotation_parts.append(f'col={self.column_start}')
+        
+        if self.column_end:
+            annotation_parts.append(f'endColumn={self.column_end}')
+
+        ','.join(annotation_parts) + '::"Ruff ({self.error_code})"'
```
Comment (dorschw) [inline riff/violation.py:47]:
```suggestion
        ','.join(annotation_parts) + f'::"Ruff ({self.error_code})"'
```

Comment (kamalq97) [reply riff/violation.py]: Fixed

File: riff/violation.py (hunk 1 above)
Comment (kamalq97) [reply riff/violation.py]: Handled edge case

**Review by dorschw — APPROVED** [review]

File: riff/violation.py [hunk 3]
```diff
@@ -30,15 +30,21 @@ class Violation(NamedTuple):
...
+        
         if self.column_start:
-            second_part += f"{self.column_start}:"
-        return f"{first_part}{second_part} {self.error_code} {self.message}"
+            annotation_parts.append(f"col={self.column_start}")
+        
+        if self.column_end:
+            annotation_parts.append(f"endColumn={self.column_end}")
+
+        return ",".join(annotation_parts) + f'::Ruff {self.error_code} ({self.message})'
```
Comment (dorschw) [inline riff/violation.py:47]:
```suggestion
        return ",".join(annotation_parts) + f'::Ruff {self.error_code}: {self.message}'
```

**Review by dorschw — APPROVED** [review]

File: riff/tests/violation_test.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            linter_name="Ruff",
+            is_autofixable=True,
+            fix_suggestion="Use 4 spaces for indentation",
+            line_end=12,
+            column_start=5,
+            column_end=20,
+        ),
+        (
+            "::error file=file.py,line=10,endLine=12,col=5,endColumn=20::"
+            'Ruff E123 (Indentation error)'
```
Comment (dorschw) [inline riff/tests/violation_test.py:26]:
```suggestion
            'Ruff E123: Indentation error'
```

File: riff/tests/violation_test.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    pytest.param(
+        Violation(
+            error_code="E123",
+            path=Path("/path/to/file.py"),
+            line_start=10,
+            message="Indentation error",
+            linter_name="Ruff",
+            is_autofixable=True,
+        ),
+        '::error file=file.py,line=10::Ruff E123 (Indentation error)',
```
Comment (dorschw) [inline riff/tests/violation_test.py:39]:
```suggestion
        '::error file=file.py,line=10::Ruff E123: Indentation error',
```

**Pull request merged**